drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -474,12 +480,15 @@ def _mark_complete_and_publish_grade(self):
         # don't publish the grade if the student has already completed the problem
         if not self.completed or (self.mode == self.ASSESSMENT_MODE and not self.attemps_remain):
             self.completed = self._is_finished() or not self.attemps_remain
-            self._publish_grade()
+            grade = self._get_grade()
+            if grade > self.max_grade:
+                self.max_grade = grade
+                self._publish_grade(grade)
 
-    def _publish_grade(self):
+    def _publish_grade(self, grade):
         try:
             self.runtime.publish(self, 'grade', {
-                'value': self._get_grade(),
+                'value': grade,
```

Comment:
Nit: will we ever want to publish a grade different from `self.max_grade`, ie. instead of the `_publish_grade` method taking a `grade` parameter, can it just publish `self.max_grade`?